kubecontrolplane/v1/types.go

```diff
@@ -184,3 +184,30 @@ type AggregatorConfig struct {
 	// proxyClientInfo specifies the client cert/key to use when proxying to aggregated API servers
 	ProxyClientInfo configv1.CertInfo `json:"proxyClientInfo" protobuf:"bytes,1,opt,name=proxyClientInfo"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type KubeControllerManagerConfig struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// serviceServingCert provides support for the old alpha service serving cert signer CA bundle
+	ServiceServingCert ServiceServingCert `json:"serviceServingCert" protobuf:"bytes,9,opt,name=serviceServingCert"`
+
+	// projectConfig is an optimization for the daemonset controller
+	ProjectConfig KubeControllerManagerProjectConfig `json:"projectConfig" protobuf:"bytes,10,opt,name=projectConfig"`
+
+	// extendedArguments is used to configure the kube-controller-manager
+	ExtendedArguments map[string]Arguments `json:"extendedArguments" protobuf:"bytes,14,rep,name=extendedArguments"`
```

Comment:
should we normalize the proto indices? or even remove them for config?

Reply:
> should we normalize the proto indices? or even remove them for config?

Yeah, we should probably remove them.  It causes us to lose schema information because of how the scripts are written.